apps/worker/src/blockchain/extrinsic.ts

```diff
@@ -63,54 +62,24 @@ export class Extrinsic<T extends ISubmittableResult = ISubmittableResult, C exte
     return this.event;
   }
 
-  public signAndSend(nonce?: number): Promise<ParsedEventResult> {
-    return firstValueFrom(
-      this.extrinsic.signAndSend(this.keys, { nonce }).pipe(
-        filter(({ status }) => status.isInBlock || status.isFinalized),
-        this.parseResult(this.event),
-      ),
+  public signAndSend(nonce?: number): Promise<[Hash, EventMap]> {
+    return firstValueFrom(this.extrinsic.signAndSend(this.keys, { nonce }).pipe(filter(({ status }) => status.isInBlock || status.isFinalized))).then(
+      ({ status, events, txHash }) => {
+        if (status.isInBlock || status.isFinalized) {
```

Comment:
doesn't this going to have to wait for it to be included in the block? If it does I think might need to get back to this later.

Reply:
I am testing this right now, but will take me longer to set things up, will cover it in next story